src/communication_interface.cc

```diff
@@ -451,22 +484,35 @@ void CommunicationInterface::HandlePause(
 void CommunicationInterface::HandleSimDriverEventTrigger(
     const ::lcm::ReceiveBuffer*, const std::string&,
     const robot_msgs::pause_cmd* cmd_msg) {
-  // bool desired_state {cmd_msg->data};
+  bool desired_state {cmd_msg->data};
   auto desired_event {cmd_msg->source};
 
   // simulate control exception
-  if (desired_event == "control_exception") {
+  if(desired_event == "control_exception") {
     dexai::log()->error(
-        "CommunicationInterface:HandleSimDriverEventTrigger: received command "
+        "CommunicationInterface::HandleSimDriverEventTrigger: received command "
         "to simulate control exception!");
     sim_control_exception_triggered_ = true;
     return;
+  } 
+  if(desired_event == "u_stop"){
+
+    if(robot_data_.robot_state.robot_mode != franka::RobotMode::kUserStopped){
+      dexai::log()->info(
+          "CommunicationInterface::HandleSimDriverEventTrigger: received command "
+          "to simulate U Stop = True");
+      robot_data_.robot_state.robot_mode = franka::RobotMode::kUserStopped;
+    }
+    else{
+      dexai::log()->info(
+          "CommunicationInterface::HandleSimDriverEventTrigger: received command "
+          "to simulate U Stop = False");
+      robot_data_.robot_state.robot_mode = franka::RobotMode::kIdle;
+    }
   } else {
     dexai::log()->error(
-        "CommunicationInterface:HandleSimDriverEventTrigger: Unrecognized "
+        "CommunicationInterface::HandleSimDriverEventTrigger: Unrecognized "
```

Comment:
Use only one colon in log messages between class and function, here and elsewhere